ruby_on_rails/advanced_forms_and_activerecord/forms_advanced.md

```diff
@@ -172,10 +172,10 @@ Sometimes Rails helper methods will do it for you, but make sure you know what y
 ### Assignment
 
 <div class="lesson-content__panel" markdown="1">
-  1. Read the [Rails Guide on Forms](https://guides.rubyonrails.org/form_helpers.html#choices-from-a-collection-of-arbitrary-objects) section 5, which covers populating a form with a collection of objects.
-  2. Read the [Same Rails Guide on Forms](http://guides.rubyonrails.org/form_helpers.html#building-complex-forms) section 10, which covers accepting nested form data.
-  3. Read the [Same Rails Guide on Forms](http://guides.rubyonrails.org/form_helpers.html#understanding-parameter-naming-conventions) section 8, which covers the parameter conventions for nested forms.
-  4. Read [this blog post from Peter Rhoades](https://www.createdbypete.com/2014/04/04/working-with-nested-forms-and-a-many-to-many-association-in-rails-4.html) on working with nested forms.  The example covers a lot of the things we've gone over so far, so follow along.  Also note how he does the allowing of nested attributes in Rails 4.
+  1. Read the Rails Guide on Forms section 5, which covers [populating a form with a collection of objects](https://guides.rubyonrails.org/form_helpers.html#choices-from-a-collection-of-arbitrary-objects).
+  2. Read the Same Rails Guide on Forms section 10, which covers [accepting nested form data](http://guides.rubyonrails.org/form_helpers.html#building-complex-forms).
+  3. Read the Same Rails Guide on Forms section 8, which covers the [parameter conventions for nested forms](http://guides.rubyonrails.org/form_helpers.html#understanding-parameter-naming-conventions).
+  4. Read this blog post from [Peter Rhoades on working with nested forms](https://www.createdbypete.com/2014/04/04/working-with-nested-forms-and-a-many-to-many-association-in-rails-4.html).The example covers a lot of the things we’ve gone over so far, so follow along. Also note how he does the allowing of nested attributes in Rails 4.
```

Comment:
```suggestion
  4. Read this blog post from [Peter Rhoades on working with nested forms](https://www.createdbypete.com/2014/04/04/working-with-nested-forms-and-a-many-to-many-association-in-rails-4.html). The example covers a lot of the things we’ve gone over so far, so follow along. Also note how he does the allowing of nested attributes in Rails 4.
```

Reply:
I'll add in the little space there. Also, I just noticed that when I copied the text from the spreadsheet to replace the original, the apostrophe changed. Originally it was a straight apostrophe `'` but after I copy pasted the text it became a curly apostrophe `’`. I think the curly apostrophe will still appear okay, but would you like me to change it back to a straight apostrophe so it's consistent everywhere? 

And sure I'll work on the lint errors! Though I'm not familiar with them, so it might take me some time.

Reply:
The apostrophe change isn't major since it'll still render the same on the website, and the one time it does matter, we have it covered by our linter fix scripts. If you're going to though, then changing them back to straight apostrophes would be great so it makes the diff a little cleaner to read when reviewing.

Not entirely sure why it happened - at some point, Google sheets seems to have changed all the straight apostrophes I wrote to curly ones, hence copying them into the lesson files picks up the change :shrug: They've been changed back to straight apostrophes on the spreadsheet now anyway.

Let me know if you run into any issues with running the lint and fix scripts. Looking at the errors, I think most if not all of them should be fixable via our scripts.

Reply:
@MaoShizhong I ran the scripts a few times on the [Advanced Forms lesson](https://www.theodinproject.com/lessons/ruby-on-rails-advanced-forms), and there's only 2 errors left which I think I have to manually change.
![image](https://github.com/TheOdinProject/curriculum/assets/96637616/b8859f9c-f531-4fb4-a9b5-ce1d588aa893)

Seems like it's expecting the conclusion to have a h4 heading but all the other headings are h3. Should I leave it at h3 to keep it consistent or change it to h4 heading? 

Also I don't think I understand the link fragment error. I checked out the link and it opens up to the right place. Looking at this [rule of link fragments](https://github.com/DavidAnson/markdownlint/blob/main/doc/md051.md#md051---link-fragments-should-be-valid), I guess the link has to be an actual subheading in order for it to be valid? But there's a few other links that point to some other place on the lesson page, and not to a subheading. The second and third links don't point to a subheading, they point to some other place on the page. 
![image](https://github.com/TheOdinProject/curriculum/assets/96637616/ca2a781b-037c-416f-9ae5-15e89724ea98)

Reply:
For the conclusion section, please change it to level 4, thanks!

For the link fragment, looks like it's caused by Markdownlint not recognising the `{: #accepts_nested_attributes_for}` syntax around line 107. There are many flavours of markdown and what we use to convert for our website also accepts that syntax, whereas Markdownlint doesn't accept it, hence it doesn't recognise that the `#accepts_nested_attributes_for` as an ID.

To resolve that, could you do the following 2 things:
1. Around line 107, replace `{: #accepts_nested_attributes_for}` with `<span id="accepts-nested-attributes-for></span>`
2. Replace the underscores in the knowledge check link fragment to dashes i.e. `#accepts-nested-attributes-for`

Changing `_` to `-` is just for consistency with other ID fragments in the file.

---

For reference, our website converts all headings to links that have a matching ID, which markdownlint will also convert and check for validity. Markdownlint will also recognise any IDs set on HTML tags in the document, hence swapping the `{: #... }` syntax for an IDed span makes it happy.

Reply:
Finished fixing lint errors for both Advanced Forms & Project: Flight Booker!